src/plugins/artifact/editor/ArtifactGraphEditor.test.js

```diff
@@ -54,28 +63,53 @@ describe("ArtifactGraphEditor", () => {
     );
   });
 
-  it("does not mutate the graph passed to its callback", () => {
+  it("modifies an artifact's description", () => {
     const onChange = jest.fn();
-    const component = (
-      <ArtifactGraphEditor
-        settings={{
-          githubApiToken: "123youdontneedme",
-          repoOwner: "sourcecred",
-          repoName: "artifact-tests",
-        }}
-        onChange={onChange}
-      />
+    const element = shallow(createComponent(onChange));
+    element
+      .find("input")
+      .simulate("change", {target: {value: "Root artifact!"}});
+    element.find("button").simulate("click");
+    element
+      .find("tr textarea")
+      .simulate("change", {target: {value: "for garlic, carrots, etc."}});
+    expect(onChange).toHaveBeenLastCalledWith(
+      new Graph().addNode({
+        address: artifactAddress(
+          new Graph(),
+          "sourcecred",
+          "artifact-tests",
+          "Root artifact!"
+        ),
+        payload: {
+          name: "Root artifact!",
+          description: "for garlic, carrots, etc.",
+        },
+      })
     );
-    const element = shallow(component);
-    expect(onChange).toHaveBeenCalledTimes(1);
+  });
+
+  it("does not mutate the graph passed to its callback", () => {
+    const onChange = jest.fn();
+    const element = shallow(createComponent(onChange));
     const g1 = onChange.mock.calls[0][0];
+    const g1Ref = g1.copy();
```

Comment:
nit: g1Ref is actually g1Copy 
(logically `const g1Ref = g1`)

Reply:
Miscommunication on my part. This meant "ref" as in "reference solution". I'll call it `g1Copy` instead.